src/pages/128x64x1/crsfdevice_page.c

```diff
@@ -0,0 +1,221 @@
+    else
+        param->u.text_sel = param->min_value;
+
+    CRSF_set_param(param, param->u.text_sel);
+}
+
+static void command_press(guiObject_t *obj, s8 press_type, const void *data)
+{
+    (void)obj;
+
+    if (press_type != -1) {
+        return;
+    }
+
+    crsf_param_t *param = (crsf_param_t *)data;
+
+    if (param->u.status == READY) {
+        CRSF_set_param(param, START);
+    }
+}
+
+static const char *value_textsel(guiObject_t *obj, int dir, void *data)
+{
+    (void)obj;
+    crsf_param_t *param = (crsf_param_t *)data;
+    u8 changed = 0;
+
+    param->u.text_sel = GUI_TextSelectHelper(param->u.text_sel,
+                            param->min_value, param->max_value,
+                            dir, 1, 1, &changed);
+
+    if (changed) CRSF_set_param(param, param->u.text_sel);
+    return current_text(param);
+}
+
+static void press_cb(struct guiObject *obj, s8 press_type, const void *data)
+{
+    (void)obj;
+    if (press_type != -1) {
+        return;
+    }
+
+    crsf_param_t *param = (crsf_param_t *)data;
+
+    switch (param->type) {
+    case FOLDER:
+        current_folder = param->id;
+        show_page(current_folder);
+        break;
+    case UINT8:
+    case UINT16:
+    case FLOAT:
+    case INT8:
+    case INT16:
+    case STRING:
+    case INFO:
+    case OUT_OF_RANGE:
+    default:
+        break;
+    }
+}
+
+static int row_cb(int absrow, int relrow, int y, void *data) {
+    (void)data;
+
+    crsf_param_t *param = current_param(absrow);
+
+    switch (param->type) {
+    case TEXT_SELECTION:
+        GUI_CreateLabelBox(&gui->name[relrow], LABEL_X, y,
+            EDIT_LABEL_WIDTH, LINE_HEIGHT, &LABEL_FONT,
+            crsf_name_cb, NULL, (void *)param);
+        if (param->max_value - param->min_value > 1) {
+            GUI_CreateTextSelectPlate(&gui->value[relrow].ts, EDIT_VALUE_X, y,
+                EDIT_VALUE_WIDTH, LINE_HEIGHT, &TEXTSEL_FONT,
+                NULL, value_textsel, (void *)param);
+        } else {
+            GUI_CreateButtonPlateText(&gui->value[relrow].but, EDIT_VALUE_X, y,
+                EDIT_VALUE_WIDTH, LINE_HEIGHT, &BUTTON_FONT,
+                crsf_value_cb, button_press, (void *)param);
+        }
+        break;
+    case COMMAND:
+        GUI_CreateLabelBox(&gui->name[relrow], LABEL_X, y,
+            LABEL_WIDTH, LINE_HEIGHT, &LABEL_FONT,
+            crsf_name_cb, command_press, (void *)param);
+        break;
+    case INFO:
+        GUI_CreateLabelBox(&gui->name[relrow], LABEL_X, y,
+            EDIT_LABEL_WIDTH, LINE_HEIGHT, &LABEL_FONT,
+            crsf_name_cb, NULL, (void *)param);
+        GUI_CreateLabelBox(&gui->value[relrow].lbl, EDIT_VALUE_X, y,
+            EDIT_VALUE_WIDTH, LINE_HEIGHT, &LABEL_FONT,
+            crsf_value_cb, NULL, (void *)param);
+        break;
+    case FOLDER:
+        GUI_CreateLabelBox(&gui->name[relrow], LABEL_X, y,
+            LABEL_WIDTH, LINE_HEIGHT, &LABEL_FONT, crsf_name_cb, press_cb, (void *)param);
+        break;
```

Comment:
Is this really different to default case?

Reply:
The default was eliminated in b863acf1 by handling all the real case values.